packages/dds/tree/src/test/simple-tree/api/schemaFactory.spec.ts

```diff
@@ -1027,6 +1028,25 @@ describe("schemaFactory", () => {
 		assert.deepEqual(getKeys(arr), [0]);
 		assert.deepEqual(getKeys(mapNode), ["x"]);
 	});
+
+	it("withMetadata", () => {
+		const factory = new SchemaFactory("");
+
+		const fooMetadata = {
+			description: "An array of numbers",
+			custom: {
+				baz: true,
+			},
+		};
+
+		class Foo extends withMetadata(factory.array("Foo", factory.number), fooMetadata) {}
+
+		assert.deepEqual(Foo.metadata, fooMetadata);
+
+		// Ensure the typing is as we expect
+		const description = Foo.metadata.description;
+		const baz = Foo.metadata.custom.baz;
```

Comment:
Not sure how this accomplishes what the comment says?

Reply:
I'll update the comment. The point is that `description` and `custom.baz` are known to TypeScript.